### PR TITLE
Fix facebook

### PR DESCRIPTION
**Pull request by tcanabrava**
While trying to fix the facebook issues described in one of the bug reports, I managed to modernize the codebase a bit. 
I didn't got hit by the bugs, whowever.

### CODE REVIEW

**Review by neolit123 — APPROVED** [review]

File: desktop-widgets/plugins/facebook/facebookconnectwidget.cpp [hunk 1]
```diff
@@ -205,13 +205,13 @@ void FacebookManager::sendDive()
 	request.setRawHeader(QByteArray("Content-Length"), QString::number(data.length()).toLocal8Bit());
 	QNetworkReply *reply = am->post(request,data);
 
-    // Why I'm blocking here?
-    // TODO: Move this so it's unblocking.
-	QEventLoop loop;
-	connect(reply, &QNetworkReply::finished, &loop, &QEventLoop::quit);
-	loop.exec();
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.cpp]: So this removes the complete changes made in the previous commit.
please combine those two when you redo the PR just to avoid the back-and-forth
(this second commit is what we should be doing in the first place, I think)

File: desktop-widgets/plugins/facebook/facebookconnectwidget.cpp [hunk 2]
```diff
@@ -282,13 +296,13 @@ SocialNetworkDialog::SocialNetworkDialog(QWidget *parent) :
...
-	connect(ui->Location, SIGNAL(clicked()), this, SLOT(selectionChanged()));
-	connect(ui->Notes, SIGNAL(clicked()), this, SLOT(selectionChanged()));
-	connect(ui->album, SIGNAL(textChanged(QString)), this, SLOT(albumChanged()));
+	connect(ui->date, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->duration, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->Buddy, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->Divemaster, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->Location, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->Notes, &QCheckBox::clicked, this, &SocialNetworkDialog::selectionChanged);
+	connect(ui->album, &QLineEdit::editingFinished, this, &SocialNetworkDialog::albumChanged);
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.cpp:325]: that conversion to the new syntax could be its own commit, right?

File: desktop-widgets/plugins/facebook/facebookconnectwidget.h [hunk 3]
```diff
@@ -24,6 +25,7 @@ class FacebookManager : public QObject
 	void requestAlbumId();
 	void requestUserId();
 	QUrl connectUrl();
+    QUrl albumListUrl();
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.h:28]: whitespace

File: desktop-widgets/plugins/facebook/facebookconnectwidget.h [hunk 4]
```diff
@@ -41,7 +43,7 @@ public slots:
 private:
 	explicit FacebookManager(QObject *parent = 0);
 	QString albumName;
-	QUrl albumListUrl;
+    QNetworkAccessManager *manager;
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.h]: whitespace

File: desktop-widgets/plugins/facebook/facebookconnectwidget.cpp [hunk 5]
```diff
@@ -162,16 +162,18 @@ void FacebookManager::requestUserId()
...
-	connect(reply, SIGNAL(finished()), &loop, SLOT(quit()));
-	loop.exec();
+	connect(reply, &QNetworkReply::finished, this, &FacebookManager::userIdReceived);
+}
 
-	QJsonDocument jsonDoc = QJsonDocument::fromJson(reply->readAll());
+void FacebookManager::userIdReceived()
+{
+    QNetworkReply *reply = qobject_cast<QNetworkReply*>(sender());
+    QJsonDocument jsonDoc = QJsonDocument::fromJson(reply->readAll());
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.cpp:169]: whitespace

File: desktop-widgets/plugins/facebook/facebookconnectwidget.h [hunk 6]
```diff
@@ -38,6 +38,7 @@ public slots:
 	void sendDive();
 	void uploadFinished();
 	void albumListReceived();
+    void userIdReceived();
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.h]: whitespace

File: desktop-widgets/plugins/facebook/facebookconnectwidget.cpp [hunk 7]
```diff
@@ -320,6 +325,15 @@ SocialNetworkDialog::SocialNetworkDialog(QWidget *parent) :
 	connect(ui->album, &QLineEdit::editingFinished, this, &SocialNetworkDialog::albumChanged);
 }
 
+SocialNetworkDialog::Size SocialNetworkDialog::profileSize() const
+{
+    QString currText = ui->profileSize->currentText();
+    return currText.startsWith(tr("Small"))  ? SMALL
+     :     currText.startsWith(tr("Medium")) ? MEDIUM
+     :  /* currText.startsWith(tr("Big")) ? */ BIG;
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.cpp:333]: whitespace

File: desktop-widgets/plugins/facebook/facebookconnectwidget.h [hunk 8]
```diff
@@ -66,9 +66,13 @@ class FacebookConnectWidget : public QDialog {
 class SocialNetworkDialog : public QDialog {
 	Q_OBJECT
 public:
+	enum Size {SMALL, MEDIUM, BIG};
+
 	SocialNetworkDialog(QWidget *parent = 0);
 	QString text() const;
 	QString album() const;
+    Size profileSize() const;
```
Comment (dirkhh) [inline desktop-widgets/plugins/facebook/facebookconnectwidget.h:74]: whitespace

**Review by dirkhh — CHANGES_REQUESTED** [review]
Tomaz, this looks really nice. I found no major problems, but quite a few small little things (one pointless commit, one commit that could be broken in two, quite a few whitespace issues)
Are you able to redo the PR? If not, I can try to fix this up manually here...
Oh, and that last commit... there's an extra 'h' in the commit message :-)

**Comment by dirkhh** [comment]
I merged this manually
